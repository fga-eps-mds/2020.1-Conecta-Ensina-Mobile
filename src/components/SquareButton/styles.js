import styled from 'styled-components'

export const Container = styled.TouchableOpacity`
background-color: ${(props) => props.theme.colors.cinzaClaro};
width: 95px;
height: 95px;
border-radius: 10px;
margin-right: 20px;
margin-bottom: 20px;
align-items: center;
`;

export const Imagem = styled.Image`
height: 50px;
width: 60px;
`;

export const Nome = styled.Text`
padding-top: 10px;
font-size: 12px;
`;