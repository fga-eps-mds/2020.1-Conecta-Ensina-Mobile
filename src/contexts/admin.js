import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';

export const AdmContext = createContext({});

function AdmProvider({children}) {
  const [students, setStudents] = useState(null);
  const [userDB, setUserDB] = useState(null);
  const [teacher, setTeacher] = useState(null);

  const {Host, user} = useContext(AuthContext);

  async function getProfessorList() {
    const fetchResponse = await fetch(`${Host}/api/student/status/0`);

    console.log(Host);

    try {
      const data = await fetchResponse.json();
      setStudents(data.data.student);
    } catch (error) {
      return error;
    }
  }

  async function statusUpdate(id) {
    const settings = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        status: 1,
        agentRole: user.role,
      }),
    };
    const fetchResponse1 = await fetch(
      `${Host}/api/student/status/${id}`,
      settings,
    );
    try {
      const data = await fetchResponse1.json();
      console.log('Success:', data);
    } catch (error) {
      console.error('Error:', error);
    }
  }

  async function getProfessoUser(id) {
    const fetchResponse = await fetch(`${Host}/api/user/${id}`);
    try {
      const data = await fetchResponse.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: data.data.user.password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      setUserDB(usuario);
      try {
        let response = await fetch(`${Host}/api/teacher/${id}`);
        const data = await response.json();
        let professor = {
          photo: data.data.teacher.photo,
          video: data.data.teacher.video,
          graduation_area: data.data.teacher.graduation_area,
          degree: data.data.teacher.degree,
          bank: data.data.teacher.bank,
          agency: data.data.teacher.agency,
          account: data.data.teacher.account,
        };
        setTeacher(professor);
      } catch (error) {
        console.error(error);
      }
    } catch (error) {
      return error;
    }
  }

  return (
    <AdmContext.Provider
      value={{
        students,
        userDB,
        teacher,
        getProfessorList,
        statusUpdate,
        getProfessoUser,
      }}>
      {children}
    </AdmContext.Provider>
  );
}

export default AdmProvider;
