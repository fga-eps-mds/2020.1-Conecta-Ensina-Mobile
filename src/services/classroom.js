export async function getNextClassroom(Host) {
  const fetchResponse = await fetch(
    `${Host}/api/classroom/nextClass/3bd7c190-ce64-4827-8c0c-58cfef45ad9f`,
  );
  try {
    const data = await fetchResponse.json();
    return data.data.classroom;
  } catch (error) {
    return error;
  }
}

export async function getUserClassroom(Host, id) {
  const fetchResponse = await fetch(`${Host}/api/classroom/userClasses/${id}`);
  try {
    const data = await fetchResponse.json();
    return data.data.classroom;
  } catch (error) {
    return error;
  }
}

export async function createClass(user, teacher, filter, student, Host) {
  const response = await fetch(`${Host}/api/classroom/create`, {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      agentRole: user.role,
      teacher: teacher,
      student: user.id,
      grade: student.grade,
      subject: filter.subject,
      dtclass: filter.dtClass,
      duration: filter.duration,
      cep: student.cep,
      number: student.number,
      details: 'oi',
    }),
  });
  try {
    const data = await response.json();
    console.log('Success: ', data);
    return data;
  } catch (error) {
    console.log(error);
  }

  console.log(response);
}
