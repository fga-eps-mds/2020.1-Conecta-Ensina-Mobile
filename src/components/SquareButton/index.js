import React from 'react';
import CustomText from '../CustomText';
import {Container, Imagem, Nome} from './styles';

export default function SquareButton({img, data, onPress, style}) {
  return (
    <Container testID="SquareButton" onPress={onPress} style={[style]}>
      <Imagem source={img} />
      <Nome>
        <CustomText black smaller>
          {data.name}
        </CustomText>
      </Nome>
    </Container>
  );
}
