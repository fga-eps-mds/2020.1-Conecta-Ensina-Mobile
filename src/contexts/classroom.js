import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Class from '../services/classroom';
import * as Student from '../services/student';
import * as Address from '../services/findAddress';
import * as GeoCode from '../services/geocode';
import {FiltersContext} from '../contexts/filters';
import {StudentContext} from '../contexts/student';

export const ClassroomContext = createContext({});

export default function ClassroomProvider({children}) {
  const [classroom, setClassroom] = useState({});
  const [firstClass, setFirstClass] = useState(null);
  const [classes, setClasses] = useState(null);
  const {Host, user} = useContext(AuthContext);
  const {stackStudents} = useContext(StudentContext);
  const {filter} = useContext(FiltersContext);
  const [statusClasses, setStatusClasses] = useState([]);
  const [statusClass, setStatusClass] = useState({});
  const [coordenates, setCoordenates] = useState({});

  async function loadNextClass() {
    const response = await Class.getNextClassroom(Host);
    if (classroom !== response) {
      setFirstClass(response);
    }
  }
  async function createClass(teacher) {
    //console.log('Esse aqui Ã© o professor: ' + teacher);
    const student = await Student.getStudent(Host, user.id);
    const response = await Class.createClass(
      user,
      teacher,
      filter,
      student.student,
      Host,
    );
  }

  async function loadUserClasses() {
    const response = await Class.getUserClassroom(Host, user.id);
    if (classroom !== response) {
      setClassroom(response);
    }
  }

  async function getClassroom(status) {
    let teacher, student;
    if (user.role === 2) {
      student = '';
      teacher = user.id;
    } else {
      student = user.id;
      teacher = '';
    }
    const response = await Class.getClassroom(student, teacher, status, Host);

    if (classroom !== response) {
      //console.log(response);
      setStatusClasses(response);
    }
  }

  async function updateStatusClassroom(id) {
    let response = await Class.readClass(Host, id);
    //console.log(response.status);
    if (response.status === 1) {
      response = await Class.updateStatusClassroom(id, 2, Host);
      //console.log(response.message);
    } else if (response.status === 2) {
      response = await Class.updateStatusClassroom(id, 3, Host);
      //console.log(response.message);
    } else if (response.status === 3) {
      response = await Class.updateStatusClassroom(id, 4, Host);
      //console.log(response.message);
    } else if (response.status === 4) {
      response = await Class.updateStatusClassroom(id, 5, Host);
      //console.log(response.message);
    }
  }
  async function loadStatusClasses(status) {
    const response = await Class.getStatusClassroom(Host, user.id, status);
    if (statusClasses !== response) {
      await setStatusClasses(response);
    }
    //console.log(response);
    return response;
  }

  async function loadStatusClassesStudents(status) {
    const response = await loadStatusClasses(status);
    let studentList = await {list: []};
    await Promise.all(
      response.map(async (item) => {
        const student = await Student.getStudent(Host, item.student);
        studentList.list = await studentList.list.concat(student);
      }),
    );
    await stackStudents(studentList.list);
  }

  async function readClass(id) {
    const response = await Class.readClass(Host, id);
    const address = await Address.findAddress(response.cep);
    let responseClass = {
      id: response.role,
      status: response.status,
      teacher: response.teacher,
      student: response.student,
      grade: response.grade,
      subject: response.subject,
      dtclass: response.dtClass,
      duration: response.duration,
      cep: response.cep,
      number: response.number,
      details: response.details,
      address: address,
    };

    setClassroom(responseClass);
  }

  async function getClass() {
    const response = await Class.getClass(Host, user);
    setClasses(response);
  }

  async function updateStatusClasses(id, status) {
    const response = await Class.updateStatus(id, status, Host);
    if (classroom !== response) {
      setClassroom(response);
    }
  }

  async function geoCode(cep) {
    const response = await GeoCode.geocode(cep);
    if (coordenates !== response) {
      //console.log(response.results[0].geometry.location);
      setCoordenates(response.results[0].geometry.location);
    }
  }

  return (
    <ClassroomContext.Provider
      value={{
        classroom,
        firstClass,
        classes,
        loadNextClass,
        createClass,
        loadUserClasses,
        getClassroom,
        updateStatusClassroom,
        loadStatusClasses,
        readClass,
        statusClasses,
        statusClass,
        setStatusClass,
        setStatusClasses,
        Host,
        getClass,
        updateStatusClasses,
        loadStatusClassesStudents,
        geoCode,
        coordenates,
      }}>
      {children}
    </ClassroomContext.Provider>
  );
}
