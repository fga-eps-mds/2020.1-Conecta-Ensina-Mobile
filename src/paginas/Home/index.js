import React, { useState } from 'react';
import Theme from '../../../Theme';
import SquareButton from '../../components/SquareButton';
import {
  WhiteContainer,
  BlueContainer,
  Usercontainer,
  UserTexto,
  UserAvatar,
} from '../../components/StyleBackground1/styles';
import { ListFiltro } from './styles'

export default function Home() {
  const [filtros, setFiltros] = useState([
    {key: '1', nome: 'Reforço Escolar', img: require('../../assets/books.png')},
    {key: '2', nome: 'Reforço Escolar', img: require('../../assets/books.png')},
    {key: '3', nome: 'Reforço Escolar', img: require('../../assets/books.png')},
  ])

  return (
    <Theme>
      <BlueContainer>
        <Usercontainer>
          <UserAvatar /> 
          <UserTexto>Nome do Aluno</UserTexto>
        </Usercontainer>
        <WhiteContainer>
          <ListFiltro 
            horizontal
            data={ filtros }
            keyExtractor={ item => item.key}
            renderItem={({ item }) => ( <SquareButton data={item} /> )}
          />
        </WhiteContainer>
      </BlueContainer>
    </Theme>
  );
}
