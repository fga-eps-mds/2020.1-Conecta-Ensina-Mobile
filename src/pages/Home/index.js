import React, {useState, useEffect, useContext} from 'react';
import Theme, {theme} from '../../../Theme';
import SquareButton from '../../components/SquareButton';
import Background1 from '../../components/Background1';
import CustomText from '../../components/CustomText';
import {ClassroomContext} from '../../contexts/classroom';

import {
  BigTextContainer,
  ButtonAulaUrgente,
  ButtonMarcarAula,
  ContainerButtons,
  ContainerAula,
  ContainerHorizontal,
  Icon,
  ListFiltro,
} from './styles';

export default function Home({navigation}) {
  const [filtros] = useState([
    {id: '1', name: 'Reforço Escolar', img: require('../../assets/books.png')},
    {id: '2', name: 'Idiomas', img: require('../../assets/books.png')},
    {id: '3', name: 'Vestibular', img: require('../../assets/books.png')},
  ]);
  const {classroom, loadNextClass, firstClass, loadUserClasses} = useContext(
    ClassroomContext,
  );
  const [selectedId, setSelectedId] = useState(null);
  const [] = useState(null);

  useEffect(() => {
    if (firstClass !== {}) {
      loadNextClass();
    }
    if (classroom !== {}) {
      loadUserClasses();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <Theme>
      <Background1 navigation={navigation} page={'Profile'}>
        <ListFiltro
          testID="listFiltro"
          horizontal
          data={filtros}
          extraData={selectedId}
          keyExtractor={(item) => item.id}
          renderItem={({item}) => {
            const backgroundColor =
              item.id === selectedId
                ? theme.colors.azulClaro
                : theme.colors.cinzaClaro;
            return (
              <SquareButton
                data={item}
                onPress={() => setSelectedId(item.id)}
                img={require('../../assets/books.png')}
                style={{backgroundColor}}
              />
            );
          }}
        />
        <ContainerAula>
          <ContainerHorizontal>
            <Icon source={require('../../assets/books.png')} />
            <CustomText bigSmall>Proxima Aula</CustomText>
          </ContainerHorizontal>
          <BigTextContainer>
            <CustomText>{firstClass.dtclass}</CustomText>
          </BigTextContainer>
          <ContainerHorizontal>
            <CustomText bigSmall>16 - 18 Horas</CustomText>
            <CustomText bigSmall>Matemática</CustomText>
          </ContainerHorizontal>
        </ContainerAula>
        <ContainerButtons>
          <ButtonMarcarAula
            onPress={() => {
              navigation.navigate('ConfirmedClass'); /*}catch(error){}*/
            }}>
            <Icon source={require('../../assets/books.png')} />
            <CustomText white medium>
              Aula Marcada
            </CustomText>
          </ButtonMarcarAula>
          <ButtonAulaUrgente>
            <Icon source={require('../../assets/books.png')} />
            <CustomText white medium>
              Aula Urgente
            </CustomText>
          </ButtonAulaUrgente>
          <ButtonMarcarAula
            testID="MarcarAula"
            onPress={() => {
              navigation.navigate('Subjects'); /*}catch(error){}*/
            }}>
            <Icon source={require('../../assets/books.png')} />
            <CustomText white medium>
              Marcar Aula
            </CustomText>
          </ButtonMarcarAula>
        </ContainerButtons>
      </Background1>
    </Theme>
  );
}
