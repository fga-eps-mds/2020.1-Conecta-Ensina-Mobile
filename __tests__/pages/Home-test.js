import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Home from '../../src/pages/Home';
import {ClassroomContext} from '../../src/contexts/classroom';
import SquareButton from '../../src/components/SquareButton';

describe('Testing Home page', () => {
  const firstClass = {
    dtclass: '2020-11-12',
  };
  const loadNextClass = jest.fn();
  const loadUserClasses = jest.fn();
  const navigation = {
    navigate: jest.fn(),
  };

  test('Should renders page', () => {
    const tree = render(
      <ClassroomContext.Provider
        value={{firstClass, loadNextClass, loadUserClasses}}>
        <Home />
      </ClassroomContext.Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('Should press square buttons', () => {
    const {UNSAFE_queryAllByType} = render(
      <ClassroomContext.Provider
        value={{firstClass, loadNextClass, loadUserClasses}}>
        <Home navigation={navigation} />
      </ClassroomContext.Provider>,
    );
    const button = UNSAFE_queryAllByType(SquareButton);

    fireEvent.press(button[0]);
    fireEvent.press(button[1]);
    fireEvent.press(button[2]);
  });

  test('Should press button Marcar Aula', () => {
    const {getByTestId} = render(
      <ClassroomContext.Provider
        value={{firstClass, loadNextClass, loadUserClasses}}>
        <Home navigation={navigation} />
      </ClassroomContext.Provider>,
    );
    const button = getByTestId('MarcarAula');

    fireEvent.press(button);
  });
});
