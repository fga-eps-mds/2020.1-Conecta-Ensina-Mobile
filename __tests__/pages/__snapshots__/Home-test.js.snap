// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home page Should renders page 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#184E79",
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "row-reverse",
        "flexGrow": 22,
        "flexShrink": 1,
        "marginLeft": 30,
        "opacity": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 1000,
          "height": 58,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginLeft": 40,
          "opacity": 1,
          "width": 58,
        }
      }
      testID="userAvatar"
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/user_blue.png",
          }
        }
        style={
          Array [
            Object {
              "height": 38,
              "width": 33,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 40,
            "marginRight": 35,
          },
        ]
      }
    >
      <Text
        medium={true}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Roboto",
              "fontSize": 25,
              "textAlign": "center",
            },
          ]
        }
        white={true}
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 78,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "id": "1",
            "img": Object {
              "testUri": "../../../src/assets/books.png",
            },
            "name": "Reforço Escolar",
          },
          Object {
            "id": "2",
            "img": Object {
              "testUri": "../../../src/assets/books.png",
            },
            "name": "Idiomas",
          },
          Object {
            "id": "3",
            "img": Object {
              "testUri": "../../../src/assets/books.png",
            },
            "name": "Vestibular",
          },
        ]
      }
      disableVirtualization={false}
      extraData={null}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "paddingBottom": "120%",
            "position": "absolute",
          },
        ]
      }
      testID="listFiltro"
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#D9D9D9",
                "borderRadius": 10,
                "height": 95,
                "marginBottom": 20,
                "marginRight": 20,
                "opacity": 1,
                "width": 95,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/books.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 50,
                    "marginTop": 8,
                    "width": 60,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "paddingTop": 10,
                  },
                ]
              }
            >
              <Text
                black={true}
                smaller={true}
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontFamily": "Roboto",
                      "fontSize": 12,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Reforço Escolar
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#D9D9D9",
                "borderRadius": 10,
                "height": 95,
                "marginBottom": 20,
                "marginRight": 20,
                "opacity": 1,
                "width": 95,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/books.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 50,
                    "marginTop": 8,
                    "width": 60,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "paddingTop": 10,
                  },
                ]
              }
            >
              <Text
                black={true}
                smaller={true}
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontFamily": "Roboto",
                      "fontSize": 12,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Idiomas
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#D9D9D9",
                "borderRadius": 10,
                "height": 95,
                "marginBottom": 20,
                "marginRight": 20,
                "opacity": 1,
                "width": 95,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/books.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 50,
                    "marginTop": 8,
                    "width": 60,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "paddingTop": 10,
                  },
                ]
              }
            >
              <Text
                black={true}
                smaller={true}
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontFamily": "Roboto",
                      "fontSize": 12,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Vestibular
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#D9D9D9",
            "borderRadius": 20,
            "height": "33%",
            "marginBottom": "15%",
            "marginTop": "15%",
            "width": "78%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 3,
              "flexShrink": 1,
              "justifyContent": "space-around",
              "paddingBottom": "2%",
              "paddingLeft": "2%",
              "paddingTop": "6%",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/books.png",
            }
          }
          style={
            Array [
              Object {
                "height": 24,
                "marginRight": "3%",
                "width": 28,
              },
            ]
          }
        />
        <Text
          bigSmall={true}
          style={
            Array [
              Object {
                "color": "#596C7C",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "textAlign": "center",
              },
            ]
          }
        >
          Proxima Aula
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginRight": "30%",
              "paddingBottom": "3%",
            },
          ]
        }
      >
        <Text
          big={true}
          style={
            Array [
              Object {
                "color": "#596C7C",
                "fontFamily": "Roboto",
                "fontSize": 45,
                "textAlign": "center",
              },
            ]
          }
        >
          12 Nov
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 3,
              "flexShrink": 1,
              "justifyContent": "space-around",
              "paddingBottom": "2%",
              "paddingLeft": "2%",
              "paddingTop": "6%",
            },
          ]
        }
      >
        <Text
          bigSmall={true}
          style={
            Array [
              Object {
                "color": "#596C7C",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "textAlign": "center",
              },
            ]
          }
        >
          16 - 18 Horas
        </Text>
        <Text
          bigSmall={true}
          style={
            Array [
              Object {
                "color": "#596C7C",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "textAlign": "center",
              },
            ]
          }
        >
          Matemática
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "height": "25%",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E46270",
            "borderRadius": 10,
            "display": "flex",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 200,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/books.png",
            }
          }
          style={
            Array [
              Object {
                "height": 24,
                "marginRight": "3%",
                "width": 28,
              },
            ]
          }
        />
        <Text
          medium={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 25,
                "textAlign": "center",
              },
            ]
          }
          white={true}
        >
          Aula Urgente
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#184E79",
            "borderRadius": 10,
            "display": "flex",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "marginTop": "7.3%",
            "opacity": 1,
            "width": 200,
          }
        }
        testID="MarcarAula"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/books.png",
            }
          }
          style={
            Array [
              Object {
                "height": 24,
                "marginRight": "3%",
                "width": 28,
              },
            ]
          }
        />
        <Text
          medium={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 25,
                "textAlign": "center",
              },
            ]
          }
          white={true}
        >
          Marcar Aula
        </Text>
      </View>
    </View>
  </View>
</View>
`;
